### PR TITLE
Adds basic search page

### LINKED ISSUE #39: Build search results page
Build out PageSearch.js.

We may want to split this into multiple issues for the filters and "Inside this Collection" features

### PR DESCRIPTION
**Pull request by helrond**
Adds the basics of a search page. It is now possible to type in a search term and get a list of results. Fixes #39 

There are some outstanding things that will be addressed in separate issues/PRs:
- Passing the query and category values to the search input #90
- Filter modal #91 
- Collection results modal #92
- Results sorting #93
- Results pagination #94

If other missing functionality is discovered, additional issues should be added.

### CODE REVIEW

File: src/components/PageSearch/index.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+              <div className="select__sort--wrapper">
+                <SelectInput className="hide-label select__sort" id="sort" label="Sort search results">
+                  <SelectOption value="" label="Sort by relevance" />
+                  <SelectOption value="title" label="Sort by title" />
+                  <SelectOption value="creator" label="Sort by creator" />
+                </SelectInput>
+              </div>
+            </div>
+            { this.state.inProgress ? (<p>Searching</p>) : (<TileList items={this.state.items} />)}
+          </main>
```
Comment (HaSistrunk) [inline src/components/PageSearch/index.js]:
```suggestion
          <div className="search-results">
            <h1 className="search__title">{`Search Results ${this.state.query && `for “${this.state.query}”` }`}</h1>
            <p className="results__summary">
              {`${this.state.startItem === this.state.endItem ?
                this.state.startItem :
                `${this.state.startItem}-${this.state.endItem}`} of ${this.state.resultsCount} results`}
            </p>
            <div className="search__controls">
              <Button
                onClick={this.toggleFacets}
                label="Filters"
                iconBefore="filter_alt"
                className="btn--filter" />
              <div className="select__sort--wrapper">
                <SelectInput className="hide-label select__sort" id="sort" label="Sort search results">
                  <SelectOption value="" label="Sort by relevance" />
                  <SelectOption value="title" label="Sort by title" />
                  <SelectOption value="creator" label="Sort by creator" />
                </SelectInput>
              </div>
            </div>
            { this.state.inProgress ? (<p>Searching</p>) : (<TileList items={this.state.items} />)}
          </div>
```

Comment (HaSistrunk) [reply src/components/PageSearch/index.js]: We currently have two `<main>`s. Do you think this change makes sense, or should we keep the search out of the main section? I realize this is also a thing on the agent pages, with the "Back to Search" button currently part of `<main>`.

**Comment by HaSistrunk** [comment]
I think this looks really good. I think we may want to make some iterative improvements to the html structure of the tiles, because although I think it works fine on the agent page to have everything in the tile as part of the link, it gets pretty verbose as a link in the search results, especially once we add the "# matches" interactivity into the mix. I'll add an issue for that.

**Comment by helrond** [comment]
Sounds good - I think there are issues around the link text and the clickable area with those tiles that we'll need to test.

**Review by HaSistrunk — APPROVED** [review]

**Pull request merged**